src/algebra/homology/chain_complex.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.graded_object
+import category_theory.differential_object
+
+/-!
+# Chain complexes
+
+We define a chain complex in `V` as a differential `ℤ`-graded object in `V`.
+
+This is fancy language for the obvious definition,
+and it seems we can use it straightforwardly:
+
+```
+example (C : chain_complex V) : C.X 5 ⟶ C.X 6 := C.d 5
+```
+
+We define the forgetful functor to `ℤ`-graded objects, and show that
+`chain_complex V` is concrete when `V` is, and `V` has coproducts.
+-/
+
+universes v u
+
+open category_theory
+open category_theory.limits
+
+section
+variables (V : Type u) [𝒱 : category.{v} V]
+include 𝒱
+
+variables [has_zero_morphisms.{v} V]
+
+/--
+A chain complex in `V` is "just" a differential `ℤ`-graded object in `V`.
+-/
+-- For now the "shift" is fixed to be the +1 direction,
+-- making this a cochain complex, rather than a chain complex.
+def chain_complex : Type (max v u) :=
```

Comment:
You seem to spend quite a bit of time repeating things you already have in `differential_object`. This is probably not a good idea, but what about changing `chain_complex` from a `def` into an `abbreviation`?

Reply:
`abbreviation` seems to work great. Let's run with that and see if we get into trouble.